File: Dhiva/moviecatalogue/app/src/main/java/id/dhiva/moviecatalogue/data/di/Injection.java
package id.dhiva.moviecatalogue.data.di;

import android.app.Application;

import id.dhiva.moviecatalogue.data.DataHelper;
import id.dhiva.moviecatalogue.data.DataRepository;

public class Injection {
    public static DataRepository provideRepository(Application application){
        return DataRepository.getInstance(new DataHelper(application));
    }
}

File: FinalSubmission/app/src/main/java/cikal/dicoding/finalsubmission/di/Injection.kt
package cikal.dicoding.finalsubmission.di

import android.content.Context
import cikal.dicoding.finalsubmission.data.source.CatalogueRepository
import cikal.dicoding.finalsubmission.data.source.remote.RemoteDataSource
import cikal.dicoding.finalsubmission.utils.JsonHelper

object Injection {
    fun provideRepository(context: Context): CatalogueRepository {

        val remoteDataSource = RemoteDataSource.getInstance(JsonHelper(context))

        return CatalogueRepository.getInstance(remoteDataSource)
    }
}